Sources/Classes/RSDeviceModeManager.m

```diff
@@ -151,31 +178,73 @@ - (void) makeFactoryDump:(RSMessage *)message FromHistory:(BOOL) fromHistory wit
+    RSMessage* originalMessage = nil;
+    for(RSTransformationEvent* transformationEvent in request.batch) {
+        if(transformationEvent.orderNo == orderNo) {
+            originalMessage = transformationEvent.event;
+        }
+    }
+    return originalMessage;
+}
+
+-(void) dumpTransformedEvents:(NSArray*) transformedPayloads toDestinationId:(NSString*) destinationId whereOriginalPayload:(RSTransformationRequest*) request {
+    NSString* destinationName = [self->destinationsWithTransformationsEnabled allKeysForObject:destinationId][0];
+    if(destinationName == nil)
+        return;
+    [RSLogger logDebug: [[NSString alloc] initWithFormat:@"RSDeviceModeManager: dumpTransformedEvents: Dumping back the transformed events to the destination %@", destinationName]];
+    for (NSDictionary* transformedPayload in transformedPayloads) {
+        NSString* status = transformedPayload[@"status"];
+        if(status == nil) continue;
+        if([status isEqualToString:@"200"]) {
+            NSDictionary* event = transformedPayload[@"event"];
+            if(event == nil) {
+                [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSDeviceModeManager: dumpTransformedEvents: User Dropped an event for the destination %@ in the transformation service hence ignoring it", destinationName]];
+                continue;
             }
+            RSMessage* transformedMessage = [[RSMessage alloc] initWithDict:transformedPayload[@"event"]];
+            [self dumpEvent:transformedMessage toDestinations:@[destinationName] withLogTag:@"dumpTransformedEvents"];
+            continue;
+        }
+        NSNumber* orderNo = [NSNumber numberWithInt:[transformedPayload[@"orderNo"] intValue]];
+        RSMessage * originalMessage = [self getOriginalEventWith:orderNo FromRequest:request];
+        if(originalMessage == nil) continue;
+        if(![self->destinationsAcceptingEventsOnTransformationError containsObject:destinationName]) {
+            [RSLogger logWarn:[[NSString alloc] initWithFormat:@"RSDeviceModeManager: dumpTransformedEvents: Destination %@ is not accepting events on transformation error, hence dropping the %@ event %@", destinationName, originalMessage.type, originalMessage.event]];
+            continue;
+        }
+        if ([status isEqualToString:@"410"]) {
+            [RSLogger logWarn: [[NSString alloc] initWithFormat:@"RSDeviceModeManager: dumpTransformedEvents: The configuration for destination %@ has been changed or transformations are disabled for it now, hence dumping back the original %@ event %@", destinationName, originalMessage.type, originalMessage.event]]; }
+        else {
+            [RSLogger logWarn: [[NSString alloc] initWithFormat:@"RSDeviceModeManager: dumpTransformedEvents: Transformation for destination %@ has failed, hence dumping back the original %@ event %@", destinationName, originalMessage.type, originalMessage.event]];
```

Comment:
I think it would be better, if we handle the 410 status code first and decide if the event is allowed or not in transformation error. In this way, a user would know if there is a 410 error or not.
If we do so, we need to update the log message as well.

Reply:
yeah, better idea, changed accordingly